### ISSUE TITLE
rtmpdump rejects restreamer url pattern. We need manual settings

### ISSUE THREAD

**Issue by gaamaaresosa** [datarhei/restreamer#516]
**Describe the bug**
Now restreamer rtmp url pattern seems unusual.
Eg:
rtmp://xxx.xxx.xxx.xxx/5b71b52a-fe27-4d10-836b-bb628451fc86.stream?token=stream
Usual pattern:
rtmp://xxx.xxx.xxx.xxx/live/path
(live = any word, path= any word )

When we pass current restreamer rtmp url to rtmpdumb, it rejects.
https://github.com/mstorsjo/rtmpdump

ERROR: You must specify a playpath (--playpath) or url (-r "rtmp://host[:port]/playpath") containing a playpath

**To Reproduce**

pi@raspberry:~ $ sudo /usr/bin/rtmpdump -r rtmp://xxx.xxx.xxx.xxx/5b71b52a-fe27-4d10-836b-bb628451fc86.stream?token=stream
RTMPDump v2.4
(c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
ERROR: You must specify a playpath (--playpath) or url (-r "rtmp://host[:port]/playpath") containing a playpath

**Expected behavior**
Please give an option to set our own rtmp url patten.
Eg: "rtmp://host[:port]/playpath"

**Desktop (please complete the following information):**
 - OS: Ubuntu Server 20.04 LTS
          Client: Raspberry Pi 4

**Comment by jstabenow** [comment]
Hey @gaamaaresosa 
You can set the `playpath` via the `App` field in the RTMP settings. Learn more here:
https://docs.datarhei.com/restreamer/knowledge-base/manual/system-settings/rtmp#6-app

Example:
rtmp://your-url:`rtmp_port`/`app`/stream_id?token=`token`

**Comment by ioppermann** [comment]
As of version 2.5.0 you can write the token as the last element of the path (like a streamkey), i.e. `rtmp://xxx.xxx.xxx.xxx/5b71b52a-fe27-4d10-836b-bb628451fc86.stream?token=stream` can be written as `rtmp://xxx.xxx.xxx.xxx/5b71b52a-fe27-4d10-836b-bb628451fc86.stream/stream`.

For rtmpdump to parse the URL correctly, you should specify the whole playpath with the `--playpath` option, e.g. `--playpath /5b71b52a-fe27-4d10-836b-bb628451fc86.stream?token=stream` or `--playpath /5b71b52a-fe27-4d10-836b-bb628451fc86.stream/stream`.

**Comment by gaamaaresosa** [comment]
> For rtmpdump to parse the URL correctly

Thanks ioppermann,
I found it last night and fixed locally.

**Comment by gaamaaresosa** [comment]
@jstabenow
> You can set the `playpath` via the `App` field in the RTMP settings

Thanks again and I am really sorry to post it on Bug section.

Btw, still I have the issue.
I set live at the app name.
But the stream ID is auto generated.
(25664aec-15cc-4cbb-bcda-6524816d11d7.stream?token=stream)
I need the stream ID also manually editable.
I tell you why.
If already some one is pushing the stream to restreamer with the given url.
And at mid night I need to update my server, or restreamer, or fully change OS
Then I need to set same manual url already given to someone to push.
So that auto retry mechanism from the playback sever will push correctly.
And our restreamer also receive it correctly.

But now every time it generates auto stream id, which is not possible for auto retry mechanism to continue the streaming.

Please consider it and update manual stream id option too.

Another issue already explained on my other post.

Please see this image.
Now I am stuck, the rtmp server could not be created without starting the playback server which is far at remote.

![Stream](https://user-images.githubusercontent.com/21296884/221568340-291d21a9-0ce8-404f-9ba2-0fd607310f9a.png)

**Comment by jstabenow** [comment]
Manual process IDs are currently not possible and not planned. 

1. to restore processes, you can export the Restreamer DB via docker to the filesystem of your host:
   `docker {...params} -v /core/config:/opt/core/config dockerimage`

   After that, upgrading or reinstalling is possible.

   Alternatively, familiarize yourself with the core API:   
   https://docs.datarhei.com/core/api-ffmpeg/process   
   Everything is possible here. Here you'll find a good place to start:  
   https://docs.datarhei.com/core/guides/beginner 

2. use static names for incoming RTMP streams and fetch them locally:     
    - Push `rtmp://ip/live/mystream?token=abc` from external
    - Use the incoming RTMP stream `rtmp://127.0.0.1/live/mystream?token=abc` as video input in a channel
    ![Bildschirmfoto 2023-02-27 um 17 28 35](https://user-images.githubusercontent.com/7462030/221686137-39269a7c-87ad-4d63-9c3e-4c8a0bedf1d1.png)
    ![Bildschirmfoto 2023-02-27 um 17 28 29](https://user-images.githubusercontent.com/7462030/221686144-84d8a0e6-9d5e-4497-806f-1bba1b02757a.png)

**Comment by gaamaaresosa** [comment]
First of all thanks agaiin !

> 1) Manual process IDs are currently not possible and not planned.

Backup and restore entire database for 1 rtmp:// server seems bit of long way.
Logically we must set the app and path manually for rtmp server like in wowza.
If I create a rtmp server for Class 6 and Class 7 then my url must be like
rtmp://ip:1935/live/class6
rtmp://ip:1935/live/class7
So any one can identify the respective url per class.

Imagine like below and how to find class7 from this:
rtmp://ip:1935/live/e1e55078-6c05-40e1-a75e-202e05e21f2d.stream
rtmp://ip:1935/live/d3e51074-6c03-40e3-b33e-212e26e23f2e.stream

> 2) use static names for incoming RTMP streams and fetch them locally:

I couldn't understand how this will work from self rtmp server.
I think this example for restreaming existing rtmp url to another server like youtube.

Anyhow I will try to understand.
Thanks again !

**Comment by jstabenow** [comment]
You can name the RTMP streams whatever you want. But the stream ID for the HLS/SRT stream will of course change and will always be a UUID. I understand that this is a problem.

We will only be able to solve this by assigning manual IDs. 
I'll be happy to put it on the roadmap.

**Comment by gaamaaresosa** [comment]
@jstabenow
Restreamer being very good and user friendly GUI
But I couldn't use it because of bad GUID stream ID. Bad logic.
We need short and sweet App name and Stream ID.
In one $5 cloud we manage 2 to 3 rtmp servers.
rtmp://subdomain/school/class1
rtmp://subdomain/school/class2
rtmp://subdomain/school/class3

I don't know why you see problem in changing the stream ID manual.
ID is just an ID, who ever needs secret, let them use long IDs.

And please make manual resolution settings also.
Auto probe 100% fails for our case.

Two days back I have removed Restreamer and back to ngnix with one server 3 apps.
Running stable. But no gui and all by commandline only, for which every, time I have to login to my cloud and carry the ssh key file with me. Borring...

Mid night I create rtmp servers and Morning they push video on to this. 
If I use Restreamer I have to run one dummy video streaming (player) to create rtmp servers because it will be done only after probing resolution.

Please give us better user friendly version soon.

Thanks again !

**Comment by jstabenow** [comment]
The Restreamer is open source and pull requests are welcome 👍 
Feel free to include the feature if it is that important. No one has asked for it yet. 
Otherwise, we will discuss and implement this idea when we have the time. 

All the best.
